test/wasi/wasi.status

```diff
@@ -6,3 +6,15 @@ prefix wasi
 
 [true] # This section applies to all platforms
 
+[$system==ibmi]
+# Unsupported on IBM i PASE
+test-wasi-clock_getres: SKIP
+test-wasi-getrusage: SKIP
+
+[$system==win32 || $system==android]
+# Unsupported on Windows and Android
+test-wasi-readdir: SKIP
```

Comment:
Shouldn't we use `common.skip()` in the test file instead?

Reply:
Using common.skip() is less desirable than using the status file, and should only be reserved in cases where the status file isn't powerful enough to express the skip condition IMO. For example imagine when this test becomes supported on Windows, to validate whether it works locally, people can just run it without modifying the tests, and remove the condition of the status file instead of modifying the tests to turn it on. I can't think of a use case where forcing the test to be skipped on Windows even when people are intentionally running the test directly is desirable.